python/ray/util/ml_utils/mlflow.py

```diff
@@ -0,0 +1,253 @@
+import os
+import logging
+from typing import Dict, Optional, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from mlflow.entities import Run
+    from mlflow.tracking import MlflowClient
+
+logger = logging.getLogger(__name__)
+
+
+class MLflowLoggerUtil:
+    """Util class for setting up and logging to MLflow.
+
+    Use this util for any library that needs MLflow logging/tracking logic
+    such as Ray Tune or Ray Train.
+    """
+
+    def __init__(self):
+        import mlflow
+        self._mlflow = mlflow
+        self.experiment_id = None
+
+    def setup_mlflow(
+            self,
+            tracking_uri: Optional[str] = None,
+            registry_uri: Optional[str] = None,
+            experiment_id: Optional[str] = None,
+            experiment_name: Optional[str] = None,
+            tracking_token=None,
+            create_experiment_if_not_exists: bool = True,
+    ) -> bool:
+        """
+        Sets up MLflow.
+
+        Sets the Mlflow tracking uri & token, and registry URI. Also sets
+        the MLflow experiment that the logger should use, and possibly
+        creates new experiment if it does not exist.
+
+        Args:
+            tracking_uri (str): The tracking URI for the MLflow tracking
+                server.
+            registry_uri (str): The registry URI for the MLflow model registry.
+            experiment_id (str): The id of an already existing MLflow
+                experiment to use for logging. If None is passed in
+                here and the MFLOW_EXPERIMENT_ID is not set, or the
+                experiment with this id does not exist,
+                ``experiment_name`` will be used instead. This argument takes
+                precedence over ``experiment_name`` if both are passed in.
+            experiment_name (str): The experiment name to use for logging.
+                If None is passed in here, the
+                the MLFLOW_EXPERIMENT_NAME environment variables is used to
+                determine the experiment name.
+                If the experiment with the name already exists with MLflow,
+                it will be reused. If not, a new experiment will be created
+                with the provided name if
+                ``create_experiment_if_not_exists`` is set to True.
+            create_experiment_if_not_exists (bool): Whether to create an
+                experiment with the provided name if it does not already
+                exist. Defaults to True.
+
+        Returns:
+            Whether setup is successful.
+        """
+        if tracking_token:
+            os.environ["MLFLOW_TRACKING_TOKEN"] = tracking_token
```

Comment:
Do we need this just during MLFLow setup? If so, maybe it would be a good idea to use a context manager to reset it to the original value after it's done

Reply:
No this is needed for the duration of the entire experiment.